selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/workspaces/WorkspaceProjects.java

```diff
@@ -120,13 +231,47 @@ public void selectProject(String projectName) {
         .click();
   }
 
+  public WebElement waitSearchField() {
+    return seleniumWebDriverHelper.waitVisibility(By.xpath(SEARCH_FIELD_XPATH));
+  }
+
   public void waitProjectDetailsPage() {
     try {
-      new WebDriverWait(seleniumWebDriver, ELEMENT_TIMEOUT_SEC)
-          .until(visibilityOfElementLocated(By.xpath(Locators.DELETE_PROJECT)));
+      waitSearchField();
+      waitAddNewProjectButton();
     } catch (TimeoutException ex) {
       // remove try-catch block after issue has been resolved
       fail("Known issue https://github.com/eclipse/che/issues/8931");
     }
   }
+
+  public boolean isCheckboxEnabled(String projectName) {
+    String checkboxLocator = format(CHECKBOX_BY_PROJECT_NAME_XPATH_TEMPLATE, projectName);
+    return seleniumWebDriverHelper
+        .waitVisibilityAndGetAttribute(By.xpath(checkboxLocator), "aria-checked")
+        .equals("true");
+  }
+
+  public void waitCheckboxEnabled(String... projectName) {
+    asList(projectName)
+        .forEach(
+            name ->
+                seleniumWebDriverHelper.waitSuccessCondition(driver -> isCheckboxEnabled(name)));
+  }
+
+  public void waitCheckboxDisabled(String... projectName) {
+    asList(projectName)
+        .forEach(
+            name ->
+                seleniumWebDriverHelper.waitSuccessCondition(driver -> !isCheckboxEnabled(name)));
+  }
+
+  public void clickOnCheckbox(String... projectName) {
+    asList(projectName)
+        .forEach(
+            name -> {
+              String checkboxLocator = format(CHECKBOX_BY_PROJECT_NAME_XPATH_TEMPLATE, projectName);
```

Comment:
looks like there is an error: should be `name` instead of `projectName` in the command:
`format(CHECKBOX_BY_PROJECT_NAME_XPATH_TEMPLATE, projectName)`